### PR TITLE
wire effects on inserting character into store

### PR DESCRIPTION
**Pull request by etnbrd**
This PR re-introduces the reactivity of effect for the character sheet, based on the character store newly introduced in https://github.com/brouberol/5esheets/pull/199.

### CODE REVIEW

File: dnd5esheets/front/src/store/index.ts [hunk 1]
```diff
@@ -114,6 +169,140 @@ class CharacterStore {
         })
       )
     }
+
+  private wireEffects = (character: ResolvedCharacter) => {
+    const abilityKeys = Object.keys(character.data.abilities) as Array<
+      keyof Abilities
+    >
+
+    const baseEffects = [
```
Comment (brouberol) [inline dnd5esheets/front/src/store/index.ts:178]: This is _beautiful_ and so simple to read :clap

File: dnd5esheets/front/src/store/index.ts [hunk 2]
```diff
@@ -114,6 +169,140 @@ class CharacterStore {
...
+          character
+        ),
+      },
+
+      // spell DC effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `spell_dc := 8 + data.scores[data.spellcasting_ability] + data.proficiency_bonus`,
```
Comment (brouberol) [inline dnd5esheets/front/src/store/index.ts:263]: Naive question: would writing `spell_dc := 8 + data.spell_attack_bonus` work here? Aka having an effect depend on another?

**Review by brouberol — APPROVED** [review]
👏 ✨

**Comment by brouberol** [comment]
⚠️ Opening a console in the character sheet page, I see the following error message: 

Screenshot 2023-08-09 at 07 45 25
Screenshot 2023-08-09 at 07 45 40

**Comment by brouberol** [comment]
This was due to the spell attack bonus and spell DC relying on the "old" data schema, causing it to access undefined keys:

- old: `spell_dc := 8 + data.scores[data.spellcasting_ability] + data.proficiency_bonus`
- new: `data.abilities[data.spellcasting_ability].modifier + data.proficiency_bonus`

**Comment by brouberol** [comment]
Fixes #149

File: dnd5esheets/front/src/store/index.ts (hunk 2 above)
Comment (etnbrd) [reply dnd5esheets/front/src/store/index.ts]: Ah! Yes, it would as long as the dependent is defined after the dependency.

**Pull request merged**